users/signals.py

```diff
@@ -0,0 +1,89 @@
+from django.dispatch import receiver
+from django.db.models.signals import post_save, pre_save, post_delete
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from asgiref.sync import async_to_sync
+from channels.layers import get_channel_layer
+
+from users.views import send_welcome_sms
+from users.models import User, OrganizationUser
+from users.serializers import UserSerializer
+from plio.cache import invalidate_cache_for_instance, invalidate_cache_for_instances
+
+from plio.settings import (
+    DEFAULT_FROM_EMAIL,
+)
+
+# the cache invalidate receivers must be defined before any other receiver,
+# so that the instance data in other receivers is always up to date.
+
+
+@receiver(post_save, sender=User)
+@receiver(post_delete, sender=User)
+def user_update_cache(sender, instance, created, raw, **kwargs):
+    invalidate_cache_for_instance(instance)
+
+    # invalidate cache for plios created by user
+    from plio.models import Plio
+
+    plios = Plio.objects.filter(created_by_id=instance.id)
+    invalidate_cache_for_instances(plios)
+
+
+@receiver(post_save, sender=OrganizationUser)
+@receiver(post_delete, sender=OrganizationUser)
+def organization_user_update_cache(sender, instance, **kwargs):
+    invalidate_cache_for_instance(instance.user)
+
+
+@receiver(pre_save, sender=User)
+def update_user(sender, instance: User, **kwargs):
```

Comment:
just one docstring here saying that it sends an SMS to new users signing up through mobile number.

Reply:
this receiver also sends an email. What about that?

Reply:
we won't be sending email here (that is done through Zapier).